src/contents.ts

```diff
@@ -656,4 +675,59 @@ namespace ContentsManager {
      */
     ajaxSettings?: IAjaxSettings;
   }
+
+  /**
+   * Get a path to a file given a relative path.
+   *
+   * @param relativePath - The relative path to the file.
+   *
+   * @param basePath - The optional base path of the file.  The
+   *   default is the server root path.
+   *
+   * #### Notes
+   * If the path is not contained within the server path, it will be
+   * returned unmodified.
+   * If the relativePath starts with a forward slash, the basePath will
```

Comment:
This function is really confusing. If relativePath is actually an absolute path, then ignore base path. But if it is relative, then ensure that base path is included, unless the relative path actually goes above base path, in which case we just return the relative path, which may still go above the base path, courtesy of `../` components.